pkg/event/events.go

```diff
@@ -125,3 +125,37 @@ func NewBackgroundSuccessEvent(policy, rule string, source Source, r *unstructur
 
 	return events
 }
+
+func NewPolicyExceptionEvent(engineResponse *response.EngineResponse, ruleResp *response.RuleResponse) Info {
+	var reasonBuilder, messageBuilder strings.Builder
+	defer reasonBuilder.Reset()
+	defer messageBuilder.Reset()
+
+	exceptionName, exceptionNamespace := getExceptionEventInfoFromRuleResponseMsg(ruleResp.Message)
+
+	fmt.Fprintf(&reasonBuilder, "exception %s/%s matches resource %s", exceptionNamespace, exceptionName, engineResponse.PatchedResource.GetName())
+	fmt.Fprintf(&reasonBuilder, "resource %s is excepted from rule %s due to exception %s/%s", engineResponse.PatchedResource.GetName(), ruleResp.Name, exceptionNamespace, exceptionName)
+
+	return Info{
+		Kind:      "PolicyException",
+		Name:      exceptionName,
+		Namespace: exceptionNamespace,
+		Reason:    reasonBuilder.String(),
+		Message:   messageBuilder.String(),
+	}
```

Comment:
I suggest we use the same Kind, Name, Namespace as other events:

```
		Kind:      getPolicyKind(engineResponse.Policy),
		Name:      engineResponse.PolicyResponse.Policy.Name,
		Namespace: engineResponse.PolicyResponse.Policy.Namespace,
```

The event will be created on the Policy, and we can add the exception info in the reason and message.

Reply:
The `reason` should be `PolicySkipped` as defined here: https://github.com/kyverno/kyverno/blob/03cec01fb5efc0f801eb5203103fa163a0233a81/pkg/event/reason.go#L7

Reply:
Sounds good 👍